File: app/src/main/java/com/example/myapplicationbird/GameView.java
package com.example.myapplicationbird;

import android.content.Context;
import android.view.SurfaceHolder;
import android.view.SurfaceView;

import androidx.annotation.NonNull;

public class GameView extends SurfaceView implements SurfaceHolder.Callback {

    GameThread gameThread;
    public GameView(Context context) {
        super(context);
        initView();
    }

    @Override
    public void surfaceCreated(@NonNull SurfaceHolder holder) {

        if (!gameThread.isRunning()){
            gameThread = new GameThread(holder);
            gameThread.start();
        }else{
            gameThread.start();
        }
    }

    @Override
    public void surfaceChanged(@NonNull SurfaceHolder holder, int format, int width, int height) {

    }

    @Override
    public void surfaceDestroyed(@NonNull SurfaceHolder holder) {
    if(gameThread.isRunning()){
    gameThread.setIsRunning(False);
    booleanretry = true;
    while(retry){
        try{
            gameThread.join();
            retry =false;
        }catch(InterruptedException e){
    }
}
    }
    void initView(){
        SurfaceHolder holder = getHolder();
        holder.addCallback(this);
        setFocusable(true);
        gameThread =  new GameThread(holder);
    }
}

File: app/src/main/java/com/example/myapplicationbird/GameView.kt
package com.example.myapplicationbird

import android.content.Context
import android.view.SurfaceHolder
import android.view.SurfaceView

class GameView(context: Context?) : SurfaceView(context), SurfaceHolder.Callback {
    private var gameThread: GameThread? = null

    init {
        initView()
    }

    override fun surfaceCreated(holder: SurfaceHolder) {
        if (!gameThread?.isRunning()!!) {
            gameThread = GameThread(holder)
            gameThread?.start()
        } else {
            gameThread?.start()
        }
    }

    override fun surfaceChanged(holder: SurfaceHolder, format: Int, width: Int, height: Int) {}

    override fun surfaceDestroyed(holder: SurfaceHolder) {
        if (gameThread?.isRunning() == true) {
            gameThread?.setIsRunning(false)
            var retry = true
            while (retry) {
                try {
                    gameThread?.join()
                    retry = false
                } catch (e: InterruptedException) {
                }
            }
        }
        initView()
    }

    private fun initView() {
        val holder = holder
        holder.addCallback(this)
        isFocusable = true
        gameThread = GameThread(holder)
    }
}